Cargo.toml

```diff
@@ -314,6 +314,7 @@ dprint-plugin-markdown = "=0.19.0"
 dprint-plugin-typescript = "=0.95.11"
 env_logger = "=0.11.6"
 fancy-regex = "=0.14.0"
+imara-diff = "=0.2.0"
```

Comment:
> I'm willing to fix this up if I get an OK about the general direction.

I think this sounds good. I kind of wonder if there's a diffing library that allows bailing after X many differences though as it would work well for incredibly large files. I wonder if we could contribute that to dissimilar.

Reply:
> > I'm willing to fix this up if I get an OK about the general direction.
> 
> I think this sounds good. I kind of wonder if there's a diffing library that allows bailing after X many differences though as it would work well for incredibly large files. I wonder if we could contribute that to dissimilar and if they'd take a patch that does that (maybe it's not too difficult?).

I think both dissimilar and imara expose an iterator over the patches, so I would assume that we can just stop iterating and thereby abort the computation of the diff early.

I have yet to check if my assumption is correct, though.

Reply:
Upstream work continues: https://github.com/pascalkuthe/imara-diff/pull/33